lib/lambda_ethereum_consensus/beacon/beacon_chain.ex

```diff
@@ -40,14 +43,31 @@ defmodule LambdaEthereumConsensus.Beacon.BeaconChain do
     GenServer.call(__MODULE__, :get_current_slot)
   end
 
-  @spec update_fork_choice_cache(Types.root(), Types.slot(), Types.root(), Types.epoch()) :: :ok
-  def update_fork_choice_cache(head_root, head_slot, finalized_root, finalized_epoch) do
+  @spec update_fork_choice_cache(Types.root(), Types.slot(), Checkpoint.t(), Checkpoint.t()) ::
+          :ok
+  def update_fork_choice_cache(head_root, head_slot, justified, finalized) do
     GenServer.cast(
       __MODULE__,
-      {:update_fork_choice_cache, head_root, head_slot, finalized_root, finalized_epoch}
+      {:update_fork_choice_cache, head_root, head_slot, justified, finalized}
     )
   end
 
+  @spec get_finalized_checkpoint() :: Types.Checkpoint.t()
+  def get_finalized_checkpoint do
+    %{finalized_root: root, finalized_epoch: epoch} =
+      GenServer.call(__MODULE__, :get_fork_choice_cache)
+
+    %Checkpoint{root: root, epoch: epoch}
```

Comment:
should we rather store it as a checkpoint then?

Reply:
Yes